src/test/java/org/springframework/integration/dsl/test/file/FileTests.java

```diff
@@ -307,7 +309,11 @@ public IntegrationFlow tailFlow() {
 		@Bean
 		public IntegrationFlow fileReadingFlow() {
 			return IntegrationFlows
-					.from(s -> s.file(tmpDir.getRoot()).patternFilter("*.sitest"),
+					.from(s -> s.file(tmpDir.getRoot())
+									.patternFilter("*.sitest")
+									.useWatchService(true)
```

Comment:
`WatchService` on OS X - had to increase timeout - increased to 60 seconds, actually took 12 seconds for file to show up.